docs/rfc/0003_vex_support.md

```diff
@@ -0,0 +1,210 @@
+out their output.
+
+That said, we want to add support for:
+
+* Keeping the configuration of multiple VEX Hub repositories
+
+* Support private VEX Hub repositories with credentials and auth tokens
+
+* Use VEX documents to scan images
+
+## Examples / User Stories
+
+[examples]: #examples
+
+<!---
+Examples of how the feature will be used. Interactions should show the action
+and the response. When appropriate, provide user stories in the form of "As a
+[role], I want [feature], so [that]."
+--->
+
+### User story #1
+
+As a user, I want to scan registries on my infrastructure, filtering out the 
+number of CVEs detected by SBOMbastic, removing false positives as much as possible.
+
+### User story #2
+
+As a user, I want to configure the registry scan with appropriate VEX files 
+depending on the content of the registry (e.g., test, staging, prod images), 
+so that the vulnerability report will have an accurate result.
+
+# Detailed design
+
+[design]: #detailed-design
+
+<!---
+This is the bulk of the RFC. Explain the design in enough detail for somebody
+familiar with the product to understand, and for somebody familiar with the
+internals to implement.
+
+This section should cover architecture aspects and the rationale behind
+disruptive technical decisions (when applicable), as well as corner-cases and
+warnings.
+--->
+
+This RFC introduces support for VEX files during registry scanning.
+From now on, the Registry CRD will have an additional field to refer to the 
+`VEXHub` resources we want to use to scan it.
+
+Suppose you are scanning the registry of a dedicated department in your company.
+This registry will host a specific kind of images depending on the scope of the 
+department. In order to make the scan as accurate as possible, we are going to 
+provide a new clusterwide CRD called `VEXHub`. This CRD will hold the 
+configuration of the VEX Hub repositories, so that the Registry CRD will refer 
+to these `VEXHub` resources.
+
+Since VEXHub CRD is clusterwide, this means that you can use the same 
+configuration across multiple registries.
+
+## VEXHub CRD
+
+To configure a new `VEXHub` resource, the user will need to apply a manifest 
+as follows:
+
+```yaml
+apiVersion: sbombastic.rancher.io/v1alpha1
+kind: VEXHub
+metadata:
+  name: company-vexhub
+  namespace: default
+spec:
+  url: "https://vex.company.com/"
+  enabled: true
+  authSecret:
+    name: company-vexhub-creds
+    namespace: default
+```
+
+There are a few attributes to keep in mind when configuring the `VEXHub` CRD:
+
+* `url`: is the URL of the VEX Hub repository (this is mandatory)
+
+* `enabled`: to enable/disable the VEX Hub repo (`true` by default)
+
+* `authSecret`: used to store auth secret to get access to the repo (optional)
+
+A VEX Hub repository might have authentication enforced. When that happens, 
+the `VEXHub` CRD has a `spec.authSecret` field, which is used to 
+reference a `Secret` resource that holds the authentication details.
```

Comment:
Maybe it's worth to point out the Secret being referenced must be placed in the same Namespace where the SBOMbastic stack is deployed

Reply:
agree